amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/RetrievalConfig.java

```diff
@@ -52,6 +57,11 @@
     @NonNull
     private final String applicationName;
 
+    /**
+     * Configurable functional interface to override the existing DataFetcher.
+     */
+    private Function<DataFetcherProviderConfig, DataFetcher> customDataFetcherProvider;
```

Comment:
Going through this again, why this shouldn't be passed directly as PollingConfig as we don't need datafetcher for fanoutconfig. This way we can avoid the new interface method in RetrievalSpecificConfig and changes in retrievalFactory() of this class. Also, we can prevent customers from setting datafetcher in fanout mode.